docs/usage/INVENTORY_SYNC.md

```diff
@@ -68,28 +68,40 @@ Therefore, in order to resolve the actual ids of those references in the sync pr
 
 ````java
-// Build an InventoryEntryQuery for fetching inventories from a source CTP project with all the needed references expanded for the sync
-final InventoryEntryQuery inventoryEntryQueryWithReferenceExpanded = InventoryReferenceResolutionUtils.buildInventoryQuery();
+// Build an InventoryEntryQuery for fetching inventories from a source CTP project without any references expanded for the sync:
+final InventoryEntryQuery inventoryEntryQueryWithReferences = InventoryReferenceResolutionUtils.buildInventoryQuery();
 
 // Query all inventories (NOTE this is just for example, please adjust your logic)
 final List<InventoryEntry> inventoryEntries =
     CtpQueryUtils
-        .queryAll(sphereClient, inventoryEntryQueryWithReferenceExpanded, Function.identity())
+        .queryAll(sphereClient, inventoryEntryQueryWithReferences, Function.identity())
         .thenApply(fetchedResources -> fetchedResources
             .stream()
             .flatMap(List::stream)
             .collect(Collectors.toList()))
         .toCompletableFuture()
         .join();
+````
+
+In order to transform and map the inventoryEntry, 
+Initialize [`InventoryReferenceTransformService`](https://github.com/commercetools/commercetools-sync-java/tree/master/src/main/java/com/commercetools/sync/inventories/service/InventoryReferenceTransformService.java) with `sphereClient` and cache(You can use your own cache implementation and pass the map).
+For cache implementation, you can refer an example class in the library - which implements the cache using caffeine library with an LRU based cache eviction strategy[`InMemoryReferenceIdToKeyCache`](https://github.com/commercetools/commercetools-sync-java/tree/master/src/main/java/com/commercetools/sync/commons/utils/InMemoryReferenceIdToKeyCache.java).
+Then call the transformInventoryReferences method with the `inventoryEntries` parameter as shown below:
 
-// Mapping from InventoryEntry to InventoryEntryDraft with considering reference resolution.
-final List<InventoryEntryDraft> inventoryEntryDrafts = InventoryReferenceResolutionUtils.mapToInventoryEntryDrafts(inventoryEntries);
+````java
+// Fetch(Id to key values for references) into the cache and map from InventoryEntry to InventoryEntryDraft using cache with considering reference resolution.
+final List<InventoryEntryDraft> inventoryEntryDrafts = InventoryReferenceTransformService.transformInventoryReferences(inventoryEntries);
 ````
 
+The cache here is used for a better performance. 
+Instead of expanding the references in the query for inventoryEntry resource. `InventoryReferenceTransformService` will execute a query to fetch key values for the Ids(nonCached) and store in cache. These cached id to key values then can be used by another resource for resolving its references instead of fetching from commercetools API. It turns out, having the in-memory LRU cache will improve the overall performance of the sync library and commercetools API.
```

Comment:
```suggestion
Instead of expanding the references in the query for inventoryEntry resource. `InventoryReferenceTransformService` will execute a query to fetch key-id pairs and store in cache. These cached id to key values then can be used by another resource for resolving its references instead of fetching from commercetools API. It turns out, having the in-memory LRU cache will improve the overall performance of the sync library and commercetools API.
```